src/components/VocabularyDrill/VocabularyDrill.tsx

```diff
@@ -105,12 +113,18 @@ function createFillIn(
 ) {
   const fillInWords = parseSourceAndTarget(words, showTips, AnswerModeType.FillIn, languageMode);
 
+  // Yes. I know. But not going to change the whole TypeScript H5P tooling now
+  let blanksl10n = params.blanksl10n as any;
+  for (const [key, value] of Object.entries(blanksl10n)) {
+    blanksl10n[key] = he.decode(value as string);
+  }
```

Comment:
Let's move `he.decode` out of the view files and into a shared util. Would make it both dry-er and easier to manage the types (`Object.entries `is great, however infamously under-typed 🥲).

My suggestion:

```ts
export const sanitizeRecord = <TRec extends Record<string, string>>(
  record: TRec,
): TRec => {
  const output = record;
  const entries: [keyof typeof record, string][] = Object.entries(record);

  for (const [key, value] of entries) {
    output[key] = decode(value) as TRec[keyof TRec];
  }

  return output;
};

```

Reply:
The reason I like to "destructure" imports is that at some point the build tools imported the entire module if it was imported as default (or with `* as X`). I believe rollup/Vite has improved on this, and I'm sure webpack has fixed their issues as well. Please enlighten me if you know more about this.

Reply:
I was merely lost in all that type shifting after having done this pull request working just fine but having to spend extra time and reworking things (including `Object.entries` that I would not have used otherwise) in order to make TypeScript happy. I have applied your approach. Thanks!